File: t/VB/AnnotationsRolesDemo/Common/Items.vb
' *************************************************************
' Copyright (c) 1991-2019 LEAD Technologies, Inc.              
' All Rights Reserved.                                         
' *************************************************************
Imports System.Collections.Generic
Imports System.Text
Imports Leadtools.Annotations.Engine

Namespace AnnotationsRolesDemo
   Public NotInheritable Class CustomRoles
      Private Sub New()
      End Sub

      Public Const RulersOnly As String = "Rulers Only"
   End Class

   Public Class CheckRoleItem
      Public Sub New(role As String)
         _role = role
      End Sub

      Private _role As String

      Public ReadOnly Property Role() As String
         Get
            Return _role
         End Get
      End Property

      Public Overrides Function ToString() As String
         Dim role As String
         Select Case _role
            Case AnnRoles.View
               role = "View"
               Exit Select

            Case AnnRoles.ViewAll
               role = "View All"
               Exit Select

            Case AnnRoles.Edit
               role = "Edit"
               Exit Select

            Case AnnRoles.EditAll
               role = "Edit All"
               Exit Select

            Case CustomRoles.RulersOnly
               role = "Rulers Only"
               Exit Select
            Case Else

               role = "Full Control"
               Exit Select
         End Select
         Return role
         'return base.ToString();
      End Function
   End Class

   Public Class UserItem
      Public Sub New(name As String)
         _userName = name
      End Sub
      Private _userName As String

      Public ReadOnly Property Name() As String
         Get
            Return _userName
         End Get
      End Property

      Private _roles As New AnnRoles()
      Public ReadOnly Property Roles() As AnnRoles
         Get
            Return _roles
         End Get
      End Property

      Public Overrides Function ToString() As String
         Return _userName
      End Function
   End Class

   Public Class GroupItem
      Private _roles As AnnRoles = Nothing
      Public Sub New(name As String)
         _groupName = name
         _roles = New AnnRoles()
      End Sub

      Public Sub New(name As String, roles As AnnRoles)
         _groupName = name
         _roles = roles
      End Sub

      Private _groupName As String

      Public ReadOnly Property Name() As String
         Get
            Return _groupName
         End Get
      End Property


      Public ReadOnly Property Roles() As AnnRoles
         Get
            Return _roles
         End Get
      End Property

      Public Overrides Function ToString() As String
         Return _groupName
      End Function
   End Class
End Namespace

File: t/CS/AnnotationsRolesDemo/Common/Items.cs
// *************************************************************
// Copyright (c) 1991-2019 LEAD Technologies, Inc.              
// All Rights Reserved.                                         
// *************************************************************
using System;
using System.Collections.Generic;
using System.Text;
using Leadtools.Annotations.Engine;

namespace AnnotationsRolesDemo
{
   public sealed class CustomRoles
   {
      private CustomRoles() { }

      public const string RulersOnly = "Rulers Only";
   }

   public class CheckRoleItem
   {
      public CheckRoleItem(string role)
      {
         _role = role;
      }

      private string _role;

      public string Role
      {
         get { return _role; }
      }

      public override string ToString()
      {
         string role;
         switch (_role)
         {
            case AnnRoles.View:
               role = "View";
               break;

            case AnnRoles.ViewAll:
               role = "View All";
               break;

            case AnnRoles.Edit:
               role = "Edit";
               break;

            case AnnRoles.EditAll:
               role = "Edit All";
               break;

            case CustomRoles.RulersOnly:
               role = "Rulers Only";
               break;

            default:
               role = "Full Control";
               break;
         }
         return role;//return base.ToString();
      }
   }

   public class UserItem
   {
      public UserItem(string name)
      {
         _userName = name;
      }
      string _userName;

      public string Name
      {
         get { return _userName; }
      }

      AnnRoles _roles = new AnnRoles();
      public AnnRoles Roles
      {
         get
         {
            return _roles;
         }
      }

      public override string ToString()
      {
         return _userName;
      }
   }

   public class GroupItem
   {
      AnnRoles _roles = null;
      public GroupItem(string name)
      {
         _groupName = name;
         _roles = new AnnRoles();
      }

      public GroupItem(string name, AnnRoles roles)
      {
         _groupName = name;
         _roles = roles;
      }

      string _groupName;

      public string Name
      {
         get { return _groupName; }
      }

      
      public AnnRoles Roles
      {
         get
         {
            return _roles;
         }
      }

      public override string ToString()
      {
         return _groupName;
      }
   }
}
